dds_ports/auto.py

```diff
@@ -26,24 +26,33 @@
 BUILD_SEMAPHORE = Semaphore(1)
 
 
-class SimpleGitHubAdaptingPort(NamedTuple):
-    package_id: PackageID
-    owner: str
-    repo: str
-    tag: str
-    package_json: PackageJSON
-    library_json: LibraryJSON
-    fs_transform: FSTransformFn
-    try_build: bool
+class SimpleGitAdaptingPort:
+    def __init__(self,
+                 package_id: PackageID,
+                 url: str,
+                 tag: str,
+                 package_json: PackageJSON,
+                 library_json: LibraryJSON,
+                 fs_transform: FSTransformFn,
+                 try_build: bool):
+        self.package_id = package_id
+        self.url = url
+        self.tag = tag
+        self.package_json = package_json
+        self.library_json = library_json
+        self.fs_transform = fs_transform
+        self.try_build = try_build
```

Comment:
Can this stay as a `NamedTuple`? If not, I'd recommend using a `dataclass`.

Reply:
`NamedTuple` didn't work. Something with the inheritance messed it up. `dataclass` does work, though.